mathics/builtin/patterns.py

```diff
@@ -476,21 +477,38 @@ def init(self, expr):
     def quick_pattern_test(self, candidate, test, evaluation):
         if test == "System`NumberQ":
             return isinstance(candidate, Number)
+        if test == "System`NumericQ":
+            if isinstance(candidate, Number):
+                return True
+            # Otherwise, follow the standard evaluation
         elif test == "System`RealNumberQ":
             if isinstance(candidate, (Integer, Rational, Real)):
                 return True
+            candidate = Expression(SymbolN, candidate).evaluate(evaluation)
```

Comment:
> We should add then a pi test to pytest (probably in a new test_constants.py file since this is testing `mathics.builtin.numbers.constants`

I just had a chance to go everything in more detail. master is _currently_ doing the right thing with respect for `NumericQ` and `NumberQ` for  `Pi` and `2 Pi`. Also there currently are (doc)tests for this behavior.

So nothing further needs to be done.